local-cli/automated-vs/solutions/vsSolution.js

```diff
@@ -0,0 +1,142 @@
+ /*
+  * Some descendant of this class needs to have:
+  * deploy
+  */
+
+'use-strict';
+
+const execSync = require('child_process').execSync;
+const path = require('path');
+const chalk = require('chalk');
+const fs = require('fs');
+const shell = require('shelljs');
+const child_process = require('child_process');
+const EOL = require('os').EOL;
+const nuget = require('../../nuget-exe/index');
+const MSBuild = require('../utils/msbuild');
+const MSBuildTools = MSBuild.MSBuildTools;
+const VS2017 = require('../visualStudios/vs2017');
+const VS2015OrEarlier = require('../visualStudios/vs2015OrEarlier');
+const utils = require('../utils/utility');
+
+module.exports = class VSSolution {
+    constructor(slnFile, root) {
+        this.slnFile = slnFile;
+        this.root = root;
+        this.name = this.getName(slnFile);
+        this.visualStudio = this.createVSVersion();
+        this.defaultMSBuildArgs = [
+            '/clp:NoSummary;NoItemAndPropertyList',
+            '/nologo',
+            '/p:AppxBundle=Never'
+        ];
+    }
+
+    getName(slnFile) { //get project name from sln file name
+        if (!/.sln$/.test(slnFile) || !shell.test('-e', slnFile)) {
+            throw new Error('Solution file (.sln) not found.');
+        }
+
+        return slnFile.slice(slnFile.search(/([^\\/:*?"<>]([a-zA-Z0-9_.]|(\\ )?))+.sln$/), -4);
+    }
+
+    //CONSTRUCTING STEPS
+    createVSVersion() {
```

Comment:
This should be abstracted out of `vsSolution`. The object should be injected into this class. You should have a default object strategy if the user omits to inject it.

Reply:
👍